_blog/deploying-laravel-to-kubernetes/index.md

```diff
@@ -0,0 +1,354 @@
+of containers. 
+
+__Nodes__
+
+As aforementioned, a Pod is simply an abstraction of the application's 
+configuration across its containers. A Node is the actual worker machine for 
+the Pods. A Node can be made up of more than one Pod and will, at the very least 
+run __Kubelet__, the process that manages the communication between the 
+Kubernetes master and the Node. Further to Kubelet, a Node will also run the 
+container runtime, handling the retrieval, unpacking and running of the application's image.
+
+__Services__
+
+As we've already established, Pods are disposable. They are created and 
+destroyed dynamically during several  system events such as scaling and 
+updating. So what if a Pod relies on communicating with another one? The 
+Pod's IP address may and probably will change on even non-trivial event. 
+Services solve this problem through configuration. This is yet another 
+abstraction layer in Kubernetes. They define a set of Pods and a policy by 
+which they can be accessed. Think of a Service (configuration) as a router 
+between Nodes, it 
+knows which IP addresses belong to each component, together with the 
+required ports and ensures to route to the correct addresses even once they 
+have changed due to any system event. 
+
+__Labels and Selectors__
+
+Several components in Kubernetes can be configured so that the abstraction 
+can be reused. Services are one of the components which do require this 
+configuration as without it there's no reference to how the Nodes can be 
+rebuilt using the Pods' abstraction. These configuration files are very 
+standard and made up of key/value pairs.
+
+__Kubectl__
+
+Kubectl is the command line interface application for managing Kubernetes. 
+This is to Kubernetes as what the Docker CLI is to Docker. Kubernetes 
+can also be managed through a GUI control panel which, as much as Kubectl does, 
+communicates with the Kubernetes Clusters through the API endpoints.
+
+
+__Minikube__
+
+Kubernetes can be run on any container cluster, including  custom ones built by 
+yourself. However, _probably_, the most popular destination is the Google Cloud 
+Platform (GCP) followed by Amazon Web Services (AWS). In any case, during 
+development, testing or learning it is best to do so locally, at least at 
+first. Minikube does exactly this. It mimics an orchestration platform within 
+your local workstation. Allowing us to deploy all our containers in their 
+respective Nodes locally as if they were hosted on the Cloud. This is great 
+while learning and testing new setups but also indispensable when testing a 
+production scenario without having to lease any external services on the 
+cloud until you're ready to do so, fully knowing what will be needed and when.
+
+__Contexts__
+
+So, we use Kubectl to manage our local Minikube cluster and maybe a few 
+more clusters on the cloud. These different orchestration targets are contexts 
+which we can tell Kubectl to switch between.
+
+
```

Comment:
As we discussed, perhaps we could reshuffle this section so that we unveil the concepts as we need them.

The aim of the article is not to deploy to Kubernetes, but to deploy and scale an application easily. It turns out that you can do that easily with Kubernetes.